nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -327,6 +327,16 @@ in
       };
     };
 
+    services.zfs.expandOnBoot = mkOption {
+      type = types.either types.bool (types.listOf types.str);
+      default = false;
+      example = [ "tank" "data" ];
+      description = ''
+        After importing, expand each device in the specified pools. If `true`, all
+        devices of all pools will be expanded.
```

Comment:
I'm not a big fan of this type, but the only alternative I can think of is `nullOr (listOf str)`, where `null` indicates all devices, and an empty list is the default for no devices. But that's not much better.

I think this is probably fine like this.

Reply:
Though I feel like `nullOr (listOf str)` is easier to reason about, since it collapses two states (both `false` and `[]` for no expandingv) into just one (`[]`)

Reply:
I definitely don't want to make `null` mean all devices, because I do want users to be able to turn it off (and have it off by default.)

Reply:
How about `either (enum [ "all" ]) (listOf str)`

Reply:
I actually have a pool named "all".

Reply:
That type would mean:
```nix
{
  # All devices
  expandOnBoot = "all";

  # The pool "all"
  expandOnBoot = [ "all" ]
}
```